middleware/middleware.js

```diff
@@ -0,0 +1,35 @@
+// middleware/route-guard.js
+const User = require('../models/User.model');
+
+// checks if the user is logged in when trying to access a specific page
+const isLoggedIn = (req, res, next) => {
+  if (req.isAuthenticated() == false) {
+    return res.redirect('/login');
+  }
+  next();
+};
+
+// if an already logged in user tries to access the login page it
+// redirects the user to the home page
+const isLoggedOut = (req, res, next) => {
+  if (req.isAuthenticated() == true) {
+    return res.redirect('/');
+  }
+  next();
+};
+
+// Checks if user has enables darkmode
+const isDarkmode = async (req, res, next) => {
```

Comment:
I don't like the name of this middleware... it's not really an "isDarkmode" but more a "provideDarkmodeInfo" + when you use it you should probably use it somewhere at the root level of the application (i.e. where you configure all the other middlewares in the config) this way *every* route will run with "darkmode" information enabled. 

but in a way you could just use the user.darkMode itself and skip the whole middleware :man_shrugging: the pass along the user as res.locals when they are logged in and have the CSS trigger in the handlebars not depend on {{#if darkmode}} but on {{#if user.darkmode}} 

which will be probably much cleaner and easier to maintain in the end.